crates/uv/src/commands/project/add.rs

```diff
@@ -93,12 +93,9 @@ pub(crate) async fn add(
         resolution_environment(python_version, python_platform, venv.interpreter())?;
 
     // Initialize the registry client.
-    let client = RegistryClientBuilder::new(cache.clone())
-        .native_tls(native_tls)
-        .connectivity(connectivity)
+    let client = RegistryClientBuilder::from(client_builder)
         .index_urls(settings.index_locations.index_urls())
         .index_strategy(settings.index_strategy)
-        .keyring(settings.keyring_provider)
```

Comment:
Nice! Would you mind auditing for additional cases where this could be used? I think there should be more.

Reply:
Good call! Found 5 more places actually!